pod-provider/frontend/src/pages/InvitePage/InvitePage.tsx

```diff
@@ -26,44 +28,50 @@ const InvitePage = () => {
   const navigate = useNavigate();
   const translate = useTranslate();
   const capabilityUri = getCapabilityUri(window.location)!;
+  const { fetch: fetchFn, getConfig } = useDataProvider<SemanticDataProvider>();
+
   const [showSuccess, setShowSuccess] = useState(false);
   const [showProviderSelect, setShowProviderSelect] = useState(false as false | 'login' | 'signup');
   const [inviterProfile, setInviterProfile] = useState(null as null | Record<string, unknown>);
   const notify = useNotify();
   const { data: identity } = useGetIdentity();
   const ownProfile = identity?.profileData;
-  const { data: capability, error: capabilityFetchError } = useGetOne('Capability', { id: capabilityUri });
 
   const outbox = useOutbox();
 
-  if (!capabilityUri) {
-    notify('app.notification.invite_cap_invalid', { type: 'error' });
-    navigate('/');
-  }
-  if (capabilityFetchError) {
-    notify('app.notification.invite_cap_fetch_error', {
-      type: 'error',
-      messageArgs: { error: JSON.stringify(capabilityFetchError) }
-    });
-    navigate('/');
-  }
-
-  // Fetch the inviter profile, once the capability document (with the inviter's profile URI) is available.
+  // Fetch the inviter profile and photo..
   useEffect(() => {
-    if (capability) {
-      fetch(capability['acl:accessTo'] as string, {
-        headers: { Accept: 'application/ld+json', Authorization: `Capability ${capabilityUri}` }
+    if (!capabilityUri) {
+      notify('app.notification.invite_cap_invalid', {
+        type: 'error',
+        multiLine: true,
+        messageArgs: { error: 'No capability available' }
+      });
+      navigate('/');
+      return;
+    }
+
+    fetchCapabilityResources(capabilityUri)
+      .then(resources => {
+        const imageBlob: Blob = resources.find(resource => resource.type?.includes('image/'));
+        const profileDoc = resources.find(resource => arrayOf(resource.type).includes('Profile'));
+
+        if (imageBlob && profileDoc) {
+          // Set the profile image as blob URI manually. Sorry, a bit hacky.
+          profileDoc['vcard:photo'] = URL.createObjectURL(imageBlob);
+        }
```

Comment:
Why do we need to do that ?

Reply:
Added a more verbose comment:
> We have to do this because setting the regular image URI would fail since the image needs to be fetched with the capability authorization.